rust/property-defs-rs/src/main.rs

```diff
@@ -65,11 +64,12 @@ fn start_health_liveness_server(config: &Config, context: Arc<AppContext>) -> Jo
 async fn spawn_producer_loop(
     consumer: Arc<StreamConsumer>,
     channel: mpsc::Sender<Update>,
+    shared_cache: Arc<Cache<Update, ()>>,
     skip_threshold: usize,
     compaction_batch_size: usize,
 ) {
-    let mut batch = Vec::with_capacity(compaction_batch_size);
-    let mut sent = HashSet::with_capacity(compaction_batch_size);
+    let mut batch = unsync::Cache::new(compaction_batch_size);
```

Comment:
Why use a Cache here? Do you actually intend for it to evict? I'm surprised it's faster than a HashSet, but if it is, maybe it's because of Rust's Hash DOS protection and we should just use a cheaper HashSet?

Mainly got worried by this line in `new`: `/// Creates a new cache with holds up to `items_capacity` items (approximately).` (emphasis on "approximately")

Reply:
It was to step around rust's slow std hash... I interpreted "approximately" to mean "can be higher", but hm.

Let me benchmark rusts HashSet

Reply:
> I interpreted "approximately" to mean "can be higher", but hm.

Fair, it could be the case.

There's also https://github.com/tkaitchuck/aHash

Reply:
It shows up, but barely compared to simply generating the set of updates. Swapped out.

Reply:
Yeah I think that's what quick_cache uses by default under the hood, just didn't reach for the other dependency. I lose about a decent chunk of event throughput in the benchmark here if I swap them out, using ahash gets me a decent improvement (baseline of ~25k/s, std::HashSet gets ~22k, AHashSet gets ~28k). I think that's worth the extra dep.